src/py21cmfast/wrapper.py

```diff
@@ -3219,3 +3224,246 @@ def calibrate_photon_cons(
             nf_estimate=neutral_fraction_photon_cons,
             NSpline=len(z_for_photon_cons),
         )
+
+
+def run_kSZ(
+    lc=None,
+    z_start=5,
+    z_end=15,
+    cosmo_params=None,
+    user_params=None,
+    astro_params=None,
+    flag_options=None,
+    PARALLEL_APPROX=False,
+    rotation=True,
+    random_seed=1,
+):
+    r"""
+    run_kSZ calculates the patchy kinetic Sunyaev-Zel'dovich signal using a 21cmFAST lighcone with density, velocity and xH_box lighcone quantities.
+
+    The function takes the cosmological quantities used in the lightcone when applicable.
+
+    Parameters
+    ----------
+    lc: :class:`~LightCone`, optional
+        lightcone object over which kSZ effect is calculated. If not provided, defalt parameters are used.
+    z_start : float, optional
+        Starting redshift for kSZ calculation, default is 5.
+    z_end : float, optional
+        Ending redshift for kSZ calculation, default is 15.
+    user_params : `~UserParams`, optional
+        Defines the overall options and parameters of the run.
+    cosmo_params : :class:`~CosmoParams`, optional
+        Defines the cosmological parameters used to compute initial conditions.
+    PARALLEL_APPROX : bool, optional
+        Flag for parrallel approximation, if True, parallel approximation is taken which is much quicker, but more approximate. Default: False.
+    rotation : bool, optional
+        Flag for rotation of boxes, if True boxes are shifted for every HII_DIM, which is the size of the simulation. Default: True.
+
+    Returns
+    -------
+    KSZOutput :
+        KSZ_box: map of the kSZ effect, in Kelvins.
+        taue_boxp: map of the optical depth.
+        l_s :  multipole moments of the power spectrum.
+        kSZ_power: Power spectrum of the kSZ effect.
+        err: Poisson error for the power spectrum.
+    """
+    if user_params is None:
+        user_params = UserParams(**UserParams._defaults_)
+    if cosmo_params is None:
+        cosmo_params = CosmoParams(**CosmoParams._defaults_)
```

Comment:
are these consistent? I assume you would like by default to use lc.user_params and lc.cosmo_params if lc is provided. Also, need to capture when cosmo_params or user_params is different between the one in lc and the one given through the arguments